include/kernel.h

```diff
@@ -1639,6 +1639,21 @@ static inline uint32_t k_cycle_get_32(void)
 	return arch_k_cycle_get_32();
 }
 
+#if defined(CONFIG_CYCLE_64BIT)
```

Comment:
I think this kind of #ifdefs in public header files are generally discouraged, as then the documentation doesn't get generated unless you add an exception to `doc/zephyr.doxyfile.in`. Just omitting the #ifdef should give an equally understandable build error if an app tries to use this API on a platform that doesn't support it.